odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1423,13 +1460,26 @@ def _validate_view(self, node, model_name, editable=True, full=False):
             node_info = {
                 'editable': editable and self._editable_node(node, name_manager),
                 'validate': validate,
+                'view_type': view_type,
             }
 
             # tag-specific validation
             validator = getattr(self, f"_validate_tag_{tag}", None)
             if validator is not None:
                 validator(node, name_manager, node_info)
 
+            # Kanban and form templates are translated into owl templates by js compilers. It is
+            # thus tempting to use owl sub components in kanban and form archs. However, this isn't
+            # something we want people to do, as those sub components are implementation details we
+            # don't want to see stored in databases.
+            elif (
+                view_type in ("form", "kanban")
+                and tag is not etree.Comment
+                and tag[0].isupper()
```

Comment:
html is case insensitive, so maybe warn here instead of raise (or check if people use uppercase)